File: src/wasteApp_client/wasteApp_mob_front/src/app/shared/models/groupment.ts
import { City } from './city';

export class Groupment {
    private id: number;
    private name: String;
    private cities: City [];
    
    
    constructor(name: String,cities: City[] ) {
        this.name = name;
        this.cities = cities;
    }
    
    
    /**
     * Getter $id
     * @return {number}
     */
	public get getId(): number {
		return this.id;
	}

    /**
     * Getter $name
     * @return {String}
     */
	public get getName(): String {
		return this.name;
	}


    /**
     * Getter $cities
     * @return {City []}
     */
	public get getCities(): City [] {
		return this.cities;
	}

    /**
     * Setter $id
     * @param {number} value
     */
	public set setId(value: number) {
		this.id = value;
	}

    /**
     * Setter $name
     * @param {String} value
     */
	public set setName(value: String) {
		this.name = value;
	}

    /**
     * Setter $cities
     * @param {City []} value
     */
	public set setCities(value: City []) {
		this.cities = value;
	}


}

File: src/wasteApp_server/src/main/java/com/fantastiSquad/wasteApp/models/entities/Groupment.java
package com.fantastiSquad.wasteApp.models.entities;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;

@Entity
@Table(name="groupment")
public class Groupment {
	
	@Id
	@GeneratedValue( strategy = GenerationType.IDENTITY)
	private Integer id;
	private String name;
	
	@OneToMany(fetch = FetchType.EAGER, cascade = {CascadeType.PERSIST})
	@Fetch(FetchMode.SELECT)
	private Set<City> city = new HashSet<>();
	
	

	public Groupment() {
		super();
	}
	
	public Groupment(String name, Set<City> city) {
		super();
		this.name = name;
		this.city = city;
	}



	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Set<City> getCity() {
		return city;
	}

	public void setCity(Set<City> city) {
		this.city = city;
	}

	@Override
	public String toString() {
		return "Groupement [id=" + id + ", name=" + name + ", city=" + city + "]";
	}
	
	

}
